goal_src/jak2/levels/common/airlock.gc

```diff
@@ -7,3 +7,1762 @@
+  (let* ((s4-0 (target-pos 0))
+         (f30-0 (com-airlock-method-25 obj s4-0 #t))
+         (s5-0 (>= (-> obj latch-open-time) (-> self clock frame-counter)))
+         )
+    (the-as
+      symbol
+      (and (or s5-0 (< (vector-vector-xz-distance (-> obj root-override trans) s4-0) (-> obj active-distance)))
+           (or s5-0 (not (-> obj y-height)) (and (>= (-> s4-0 y) (- (-> obj root-override trans y) (-> obj y-height y)))
+                                                 (< (-> s4-0 y) (+ (-> obj root-override trans y) (-> obj y-height x)))
+                                                 )
+               )
+           (begin
+             (if (and (not (-> obj were-behind?)) (and (< f30-0 0.0) (-> obj inner?)))
+                 (set! (-> obj were-behind?) #t)
+                 )
+             (and (< (-> obj latch-closed-time) (-> self clock frame-counter))
+                  (or (not (and *target* (logtest? (focus-status fs20 fs30) (-> *target* focus-status)))) (< f30-0 -409.6))
+                  (or (and (< f30-0 (-> obj open-distance))
+                           (or (not (-> obj were-behind?)) (< f30-0 20480.0))
+                           (and (or (< 409.6 f30-0)
+                                    (begin
+                                      (let ((a0-11 (-> obj entity)))
+                                        (set! a0-12 (entity-actor-lookup a0-11 'next-actor 0))
+                                        )
+                                      (not a0-12)
+                                      )
+                                    (logtest? (-> a0-12 extra perm status) (entity-perm-status subtask-complete))
+                                    )
+                                (and (script-context-method-9
+                                       (new 'stack 'script-context (the-as basic (process->ppointer self)) self (the-as vector #f))
+                                       (the-as structure (-> obj open-test))
+                                       )
+                                     (-> *setting-control* user-current airlock)
+                                     )
+                                )
+                           )
+                      s5-0
+                      (let ((f0-8 (com-airlock-method-25 obj (camera-pos) #f)))
+                        (and (or (not *target*) (zero? (logand (-> *target* focus-status) (focus-status fs8))))
+                             (or (< (* f30-0 f0-8) 0.0)
+                                 (and (< (fabs f0-8) 4096.0)
+                                      (< (vector-vector-xz-distance (camera-pos) (-> obj root-override trans)) (-> obj door-radius))
+                                      )
+                                 )
+                             )
+                        )
+                      )
+                  )
+             )
+           )
+      )
+    )
+  )
+
+(defmethod com-airlock-method-24 com-airlock ((obj com-airlock) (arg0 symbol))
```

Comment:
looks like this checks to see if the destination level is loaded loaded, maybe `destination-loaded?`